Mathlib/Data/Complex/Order.lean

```diff
@@ -63,6 +63,32 @@ theorem nonneg_iff {z : ℂ} : 0 ≤ z ↔ 0 ≤ z.re ∧ 0 = z.im :=
 theorem pos_iff {z : ℂ} : 0 < z ↔ 0 < z.re ∧ 0 = z.im :=
   lt_def
 
+theorem nonpos_iff {z : ℂ} : z ≤ 0 ↔ z.re ≤ 0 ∧ z.im = 0 :=
+  le_def
+
+theorem neg_iff {z : ℂ} : z < 0 ↔ z.re < 0 ∧ z.im = 0 :=
+  lt_def
+
+theorem ofReal_mul_pos_iff (x : ℝ) (z : ℂ) :
+    0 < x * z ↔ (x < 0 ∧ z < 0) ∨ (0 < x ∧ 0 < z) := by
+  simp_rw [pos_iff, neg_iff, re_ofReal_mul, im_ofReal_mul]
+  obtain hx | hx | hx := lt_trichotomy x 0
+  · simp_rw [mul_pos_iff, eq_comm, mul_eq_zero, hx, not_lt_of_gt hx, hx.ne, false_and, true_and,
+      false_or, or_false]
+  · simp_rw [hx, zero_mul, lt_self_iff_false, false_and, false_or]
+  · simp_rw [mul_pos_iff, eq_comm, mul_eq_zero, hx, not_lt_of_gt hx, hx.ne', false_and, true_and,
+      false_or, or_false]
+
+theorem ofReal_mul_neg_iff (x : ℝ) (z : ℂ) :
+    x * z < 0 ↔ (x < 0 ∧ 0 < z) ∨ (0 < x ∧ z < 0) := by
+  simp_rw [pos_iff, neg_iff, re_ofReal_mul, im_ofReal_mul]
+  obtain hx | hx | hx := lt_trichotomy x 0
+  · simp_rw [mul_neg_iff, mul_eq_zero, hx, not_lt_of_gt hx, hx.ne, false_and, true_and, false_or,
+      or_false, eq_comm]
+  · simp_rw [hx, zero_mul, lt_self_iff_false, false_and, false_or]
+  · simp_rw [mul_neg_iff, mul_eq_zero, hx, not_lt_of_gt hx, hx.ne', false_and, true_and, false_or,
+      or_false]
```

Comment:
untested, but something like this should work:
```suggestion
  simpa only [mul_neg, neg_pos, neg_neg_iff_pos] using ofReal_mul_pos_iff x (-z)
```

Reply:
Hum, that doesn't seem to work because the corresponding `CovariantClass ℂ ℂ` instance is missing for `neg_pos`. In fact, I don't think it holds in this case.

Reply:
@xroblot no, this definitely works, it's just that Mathlib doesn't know `ℂ` is `StrictOrderedCommRing` until `Analysis.Complex.Basic` because it gets the `RCLike.toStrictOrderedCommRing` instance. You can see this in the playground:
```
import Mathlib.Data.Complex.Order
import Mathlib.Analysis.Complex.Basic

open ComplexOrder Complex

theorem nonpos_iff {z : ℂ} : z ≤ 0 ↔ z.re ≤ 0 ∧ z.im = 0 :=
  le_def

theorem neg_iff {z : ℂ} : z < 0 ↔ z.re < 0 ∧ z.im = 0 :=
  lt_def

theorem ofReal_mul_pos_iff (x : ℝ) (z : ℂ) :
    0 < x * z ↔ (x < 0 ∧ z < 0) ∨ (0 < x ∧ 0 < z) := by
  simp_rw [pos_iff, neg_iff, re_ofReal_mul, im_ofReal_mul]
  obtain hx | hx | hx := lt_trichotomy x 0
  · simp_rw [mul_pos_iff, eq_comm, mul_eq_zero, hx, not_lt_of_gt hx, hx.ne, false_and, true_and,
      false_or, or_false]
  · simp_rw [hx, zero_mul, lt_self_iff_false, false_and, false_or]
  · simp_rw [mul_pos_iff, eq_comm, mul_eq_zero, hx, not_lt_of_gt hx, hx.ne', false_and, true_and,
      false_or, or_false]

theorem ofReal_mul_neg_iff (x : ℝ) (z : ℂ) :
    x * z < 0 ↔ (x < 0 ∧ 0 < z) ∨ (0 < x ∧ z < 0) := by
  simpa only [mul_neg, neg_pos, neg_neg_iff_pos] using ofReal_mul_pos_iff x (-z)
```
Obviously we can't import that here.

I think probably what you want is to either copy the proofs of `RCLike.toStrictOrderedCommRing` and `RCLike.toStarOrderedRing` into this file, or else prove these lemmas for `RCLike` and move them to the later file.

I would suggest the latter, unless for some reason you really only want these lemmas for `ℂ` and you really want to avoid importing analysis into wherever you use them.

Reply:
Sorry, I completely missed that point. 

Anyway, thanks for the explanation. I moved the results to `RCLike.Basic`.